inc/mailhandlers/smtp.php

```diff
@@ -247,7 +247,10 @@ function send()
 				return false;
 			}
 
-			$this->send_data('.', 250);
+			if(!$this->send_data('.', 250))
+			{
+				$this->fatal_error("Mail may not delivered. Reason: ".$this->get_error());
```

Comment:
I think the wording for this could be improved:

```
Mail may not be delivered. Reason: {$this->get_error()}
```

Otherwise, this looks like a nice simple enough change.

Reply:
Yep it's pretty simple and obvious that a code 250 is expected after the message is accepted by server.

Had a test with Zoho SMTP service and if SMTP configurations and mailing fields are correctly set according to Zoho's requirement, mails deliver flawless; if `from:` is set to a sender other than its required user for authentication, an error with 553 message is logged in System Mail Log.